coderd/provisionerdserver/provisionerdserver.go

```diff
@@ -467,6 +467,23 @@ func (s *server) acquireProtoJob(ctx context.Context, job database.ProvisionerJo
 		if err != nil {
 			return nil, failJob(fmt.Sprintf("get owner: %s", err))
 		}
+		orgGroups, err := s.Database.GetGroupsByOrganizationID(ctx, s.OrganizationID)
+		if err != nil {
+			return nil, failJob(fmt.Sprintf("get owner groups: %s", err))
+		}
+		ownerGroupNames := []string{}
+		for _, group := range orgGroups {
+			members, err := s.Database.GetGroupMembers(ctx, group.ID)
+			if err != nil {
+				return nil, failJob(fmt.Sprintf("get group members: %s", err))
+			}
+			for _, member := range members {
+				if member.ID == owner.ID {
+					ownerGroupNames = append(ownerGroupNames, group.Name)
+					break
+				}
+			}
+		}
```

Comment:
Wouldn't this be better solved with a query that specifically returns a user's groups?

Reply:
I agree and what I started with, but I saw every other place we get group members we do this pattern. I'll make a new query but I'll probably leave the other places this happens alone for this PR.

Reply:
Yeah completely fine, I agree the groups code could be improved (something I authored 🙂 )

Reply:
Using a new query now